cljfmt/src/cljfmt/core.cljc

```diff
@@ -192,6 +192,28 @@
 (defn pattern? [v]
   (instance? #?(:clj Pattern :cljs js/RegExp) v))
 
+(defn root? [zloc]
+  (nil? (zip/up zloc)))
+
+(defn- top-level-form [zloc]
+  (->> zloc
+       (iterate z/up)
+       (take-while (complement root?))
+       last))
+
+(defn- ns-form? [zloc]
+  (some-> zloc
+          child-sexprs
+          first
+          #{'ns `ns}))
```

Comment:
Maybe, except \`ns is going to be `cljfmt.core/ns`.

Let's keep it as `'ns` for now. If it turns out to be a problem, we can fix it in future. I think there's quite a few tools already that assume `ns` rather than `clojure.core/ns`.

Reply:
It actually expands to `clojure.core/ns`

Can change, just confirm

Reply:
Let's change it to `(= 'ns)` for now, because this is technically a change outside the current PR. However, I'd accept a small additional PR to change it to:

     #{'ns `ns}

Also we can put it all on one line:

```clojure
(defn- ns-form? [zloc]
  (some-> zloc z/sexpr first (= 'ns)))
```